R/REST_helpers.R

```diff
@@ -36,7 +36,12 @@ sbtools_POST <- function(url, body, ..., session){
 #' @keywords internal
 sbtools_GET <- function(url, ..., session) {
 	supported_types <- c('text/plain','text/csv','text/tab-separated-values','application/json','application/x-gzip', 'application/pdf')
-	r = GET(url = url, ..., httrUserAgent(), handle = session)
+	tryCatch({
```

Comment:
Have you been able to trigger an error and see how this does? Does this catch _any_ error and return the same thing? Such as not being authenticated, the item not existing, etc? Does it treat SB being down as the same thing as an item not existing?

Reply:
It will treat any unhandled error the same as before but add this bit of context. Note that `r` still gets passed to the more specific error handler if GET doesn't produce an error.

Reply:
And no, I haven't been able to trigger an error here. Little heartburn there, but it should work...
``` r
tryCatch({
    r = GET("borked")
}, error = function(e) stop(paste("Error when calling ScienceBase,", 
    "internet or server down? Original", "error was:\n", e)))
#> Error in value[[3L]](cond): Error when calling ScienceBase, internet or server down? Original error was:
#>  Error in GET("borked"): could not find function "GET"
```

<sup>Created on 2020-04-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Not really a useful example though since the package has access to `httr::GET` and your example doesn't. You are failing on the library issue, not the web call.

Reply:
Details... 😉 My point was that the `tryCatch` functions as much as anything.

``` r
tryCatch({
    r = httr::GET("borked")
}, error = function(e) stop(paste("Error when calling ScienceBase,", 
    "internet or server down? Original", "error was:\n", e)))
#> Error in value[[3L]](cond): Error when calling ScienceBase, internet or server down? Original error was:
#>  Error in curl::curl_fetch_memory(url, handle = handle): Could not resolve host: borked
```

<sup>Created on 2020-04-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>